blockchain/blockchain.go

```diff
@@ -758,22 +763,55 @@ func revertHead(txn db.Transaction) error {
 		return err
 	}
 
+	genesisBlock := blockNumber == 0
+
 	// remove block header
 	if err = txn.Delete(db.BlockHeadersByNumber.Key(numBytes)); err != nil {
 		return err
 	}
 	if err = txn.Delete(db.BlockHeaderNumbersByHash.Key(header.Hash.Marshal())); err != nil {
 		return err
 	}
+	if !genesisBlock {
+		var newHeader *core.Header
+		newHeader, err = blockHeaderByNumber(txn, blockNumber-1)
+		if err != nil {
+			return err
+		}
+		b.newHeads.Send(newHeader)
```

Comment:
The user can save the most recent received header `A`. Upon receiving a new header `B`, one of two things must be true:

1. `B`'s parent is `A`: `B.ParentHash.Equal(A.Hash)`
2. `B` is `A`: `B.Hash.Equal(A.Hash)`

Since it's simple for a user to distinguish reverted/new blocks, I don't think it's necessary to do anything on our end. We should make that distinction for logs, though, since there isn't a good way for a client to decide. Wdyt?